app-android/src/main/java/com/ustadmobile/port/android/view/PersonDetailFragment2.kt

```diff
@@ -79,31 +98,368 @@ class PersonDetailFragment2 : Fragment(){
+        }
+
+        if (uiState.person?.gender != 0
+            && uiState.person?.gender != null){
+            DetailFeild(0,
+                gender,
+                stringResource(R.string.gender_literal))
+        }
+
+        if (uiState.person?.personOrgId != null){
+            DetailFeild(
+                R.drawable.ic_badge_24dp,
+                uiState.person?.personOrgId ?: "",
+                stringResource(R.string.organization_id))
+        }
+
+        if (!uiState.person?.username.isNullOrEmpty()){
+            DetailFeild(
+                R.drawable.ic_account_circle_black_24dp,
+                uiState.person?.username ?: "",
+                stringResource(R.string.username))
+        }
+    }
+}
 
 @Composable
-fun PersonDetailScreen(viewModel: PersonDetailViewModel) {
+private fun ContactDetails(
+    uiState: PersonDetailUiState,
+    onClickDial: () -> Unit = {},
+    onClickSms: () -> Unit = {},
+    onClickEmail: () -> Unit = {},){
+    Column(
+        modifier = Modifier.padding(8.dp)
+    ) {
+
+        if (!uiState.person?.phoneNum.isNullOrEmpty()){
+            CallRow(
+                uiState.person?.phoneNum ?: "",
+                onClickDial,
+                onClickSms)
+        }
 
+        if (!uiState.person?.emailAddr.isNullOrEmpty()){
+            DetailFeild(
+                R.drawable.ic_email_black_24dp,
+                uiState.person?.emailAddr ?: "",
+                stringResource(R.string.email),
+                onClickEmail)
+        }
 
+        if (!uiState.person?.personAddress.isNullOrEmpty()){
+            DetailFeild(
+                R.drawable.ic_location_pin_24dp,
+                uiState.person?.personAddress ?: "",
+                stringResource(R.string.address))
+        }
+    }
+}
+
+@Composable
+private fun CallRow(
+    phoneNum: String,
+    onClickDial: () -> Unit = {},
+    onClickSms: () -> Unit = {},){
+
+    Row(
+        modifier = Modifier.fillMaxWidth(),
+        horizontalArrangement = Arrangement.SpaceBetween
+    ) {
+
+        DetailFeild(
+            R.drawable.ic_phone_black_24dp,
+            phoneNum,
+            stringResource(R.string.phone),
+            onClickDial)
+
+        TextButton(
+            onClick = onClickSms
+        ){
+            Image(
+                painter = painterResource(id = R.drawable.ic_message_bcd4_24dp),
+                contentDescription = null,
+                colorFilter = ColorFilter.tint(color = Color.Gray),
+                modifier = Modifier
+                    .size(24.dp))
+        }
+    }
+}
+
+@Composable
+private fun Classes(
+    clazzes: List<ClazzEnrolmentWithClazzAndAttendance> = emptyList(),
+    onClickClazz: () -> Unit = {}){
+
+    clazzes.forEach { clazz ->
+        Spacer(modifier = Modifier.height(15.dp))
+
+        Button(onClick = onClickClazz) {
+            ClassItem(clazz)
+        }
+    }
+}
+
+@Composable
+private fun ClassItem(clazz: ClazzEnrolmentWithClazzAndAttendance){
```

Comment:
This needs to show the same information that was shown using the Xml - e.g. list of classes.